streaq/types.py

```diff
@@ -1,13 +1,30 @@
+from __future__ import annotations
 from dataclasses import dataclass
 from datetime import timedelta
-from typing import Generic, ParamSpec, TypeVar
+from typing import (
+    Concatenate,
+    Generic,
+    ParamSpec,
+    Protocol,
+    Callable,
+    Coroutine,
+    Any,
+    TypeAlias,
+    TYPE_CHECKING,
+    overload,
+    TypeVar,
+)
 
 from coredis import Redis
 
+if TYPE_CHECKING:
+    from streaq.task import RegisteredTask, RegisteredCron
+
 P = ParamSpec("P")
 POther = ParamSpec("POther")
 R = TypeVar("R")
 ROther = TypeVar("ROther")
+RCo = TypeVar("RCo", covariant=True)
```

Comment:
Could you test and see if the existing R TypeVar can be made covariant? If that's the case, we don't need another TypeVar here. Otherwise, I'd prefer a more descriptive name.

Reply:
Done